From: Lillian Berry <lillian@star-ark.net>
Subject: init/main.c: add warning when file specified in rdinit is inaccessible
Date: Mon, 7 Jul 2025 09:14:11 +0000

Avoid silently ignoring the initramfs when the file specified in rdinit is
not usable.  This prints an error that clearly explains the issue (file
was not found, vs initramfs was not found).

Link: https://lkml.kernel.org/r/20250707091411.1412681-1-lillian@star-ark.net
Signed-off-by: Lillian Berry <lillian@star-ark.net>
Cc: Al Viro <viro@zeniv.linux.org.uk>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 init/main.c |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/init/main.c~init-mainc-add-warning-when-file-specified-in-rdinit-is-inaccessible
+++ a/init/main.c
@@ -1592,7 +1592,11 @@ static noinline void __init kernel_init_
 	 * check if there is an early userspace init.  If yes, let it do all
 	 * the work
 	 */
-	if (init_eaccess(ramdisk_execute_command) != 0) {
+	int ramdisk_command_access;
+	ramdisk_command_access = init_eaccess(ramdisk_execute_command);
+	if (ramdisk_command_access != 0) {
+		pr_warn("check access for rdinit=%s failed: %i, ignoring\n",
+			ramdisk_execute_command, ramdisk_command_access);
 		ramdisk_execute_command = NULL;
 		prepare_namespace();
 	}
_
